controllers/datadogagent/feature/helmcheck/feature.go

```diff
@@ -80,6 +85,24 @@ func (f *helmCheckFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp featur
 			f.serviceAccountName = v2alpha1.GetClusterChecksRunnerServiceAccount(dda)
 			reqComp.ClusterChecksRunner.IsRequired = apiutils.NewBoolPointer(true)
 		}
+
+		// Build configMap based on feature flags.
+		cm, err := f.buildHelmCheckConfigMap()
+		if err != nil {
+			f.logger.Error(err, "couldn't generate configMap for helm check")
+		}
+		f.config = cm
+
+		// Create hash based on configMap.
+		hash, err := comparison.GenerateMD5ForSpec(cm.Data)
+		if err != nil {
+			f.logger.Error(err, "couldn't generate hash for helm check config")
+		} else {
+			f.logger.V(2).Info("built helm check from config", "hash", hash)
+		}
+
+		f.configAnnotationValue = hash
+		f.configAnnotationKey = fmt.Sprintf("checksum/%s-config", feature.HelmCheckIDType)
```

Comment:
Should we add a new const for the helm check config annotation key? There's the `MD5ChecksumAnnotationKey = "checksum/%s-custom-config"` [const](https://github.com/DataDog/datadog-operator/blob/main/apis/datadoghq/common/const.go#L25), but it's more for custom configs

Reply:
That const is used for CC feature already and don't see problem using it here
https://github.com/DataDog/datadog-operator/blob/36c4c4a6204e9522def1d76eecc042e58f480de8/controllers/datadogagent/feature/clusterchecks/feature.go#L267-L268